api/app/db/crud/stations.py

```diff
@@ -0,0 +1,64 @@
+from datetime import datetime, timedelta, timezone
+from sqlalchemy.sql.expression import literal
+from sqlalchemy.orm import Session
+from sqlalchemy import func
+from app.db.models.forecasts import NoonForecast
+from app.db.models.observations import HourlyActual
+
+
+def _get_noon_date(date_of_interest: datetime):
+    """ 
+    If before noon today, give noon from day before.
+    If after noon today, give noon from date of interest.
+    """
+    noon_for_today = datetime(year=date_of_interest.year,
+                              month=date_of_interest.month,
+                              day=date_of_interest.day,
+                              hour=20, tzinfo=timezone.utc)
+    if date_of_interest < noon_for_today:
+        # oh - you want noon for the prevous day
+        yesterday = date_of_interest - timedelta(days=1)
+        return datetime(
+            year=yesterday.year, month=yesterday.month, day=yesterday.day, hour=20, tzinfo=timezone.utc)
+    # you want noon for today
+    return noon_for_today
```

Comment:
I think the code would be more readable if we changed the variable names `noon_for_today` -> `noon_for_toi` and `yesterday` -> `previous_day`. When I first read this I was confused because I was thinking that "yesterday" literally meant yesterday, and was imagining lots of problems when users are looking up dates of interest from previous years

Reply:
@andrea-williams have a look, I re-factored this method before I saw your comment.